### ISSUE TITLE
Skipping a player when I set a new phase

### ISSUE THREAD

**Issue by DerrickAfrifa** [boardgameio/boardgame.io#1036]
In this slightly contrived example, I allow player to add and subtract from a `deck` numeric variable. In the `onEnd` hook I check whether the deck is greater than 1 and if it is I set the phase to the `testPhase`. However, when setting the phase one player is skipped. It goes from player 1 to player 3 in this example and 2 is skipped. More specifically, the turn is started and immediately ended for player 2. Is this expected behaviour? If so can someone explain why that's the case?

Thanks

```
const add = (G, ctx) => {
  G.deck = G.deck + 1;
  ctx.events.endTurn();
}

const subtract = (G, ctx) => {
  G.deck = G.deck - 1;
  ctx.events.endTurn();
}

export const TestGame = {
  setup: (ctx, setupData) => {
    const deck = 0;
    return {
      deck,
    };
  },

  moves: {
    add,
    subtract
  },

  turn: {
    onBegin: (G, ctx) => {
      console.log("beginning turn for player", ctx.currentPlayer);
    },
    onEnd: (G, ctx) => {
      console.log("\n\n\n\nending turn for player", ctx.currentPlayer)
      if (G.deck > 1) {
        ctx.events.setPhase("testPhase");
      }
    }
  },

  phases: {
    testPhase: {
      moves: {
        subtract,
      }
    }
  }
}
```

**Comment by delucis** [comment]
@DerrickAfrifa Hmm, sounds like it might be a bug!

When `setPhase` or `endPhase` is called, we first end the turn of the current player. There should be a check so that `endTurn` followed by `setPhase` shouldn't end two players' turns but maybe this is something we've overlooked.

**Comment by DerrickAfrifa** [comment]
> @DerrickAfrifa Hmm, sounds like it might be a bug!
> 
> When `setPhase` or `endPhase` is called, we first end the turn of the current player. There should be a check so that `endTurn` followed by `setPhase` shouldn't end two players' turns but maybe this is something we've overlooked.

Yes, it looks like a bug to me. Do you have any plans of implementing a fix? I'm not sure if the library is under maintenance.

**Comment by delucis** [comment]
It’s been a little slow the last month or two due to work pressures, but a fix should be possible and there might be some time coming up. PRs are also always welcome if someone has the time. I’m not sure without digging, but the problem’s probably somewhere in [the events plugin](https://github.com/boardgameio/boardgame.io/blob/c93b316dc0731764d0f1b05026d5ccc631c9df71/src/plugins/events/events.ts#L120-L208).

Other things that might be helpful in the meantime would be a minimal Code Sandbox using your example code and/or a test case (maybe in flow
